### PR TITLE
Adding a new zram-generator package

### PR DESCRIPTION
**Pull request by george-angel**
# Adding sys-apps/zram-generator package

This resolves https://github.com/flatcar/Flatcar/issues/905 . Issue explains the advantages of adding this package, but it effectively minimises the amount of config / setup a user needs to do as well as allowing for "dynamic" sizing of swap.

## How to use

Create an image that contains this package and follow instructions at: https://wiki.gentoo.org/wiki/Zram#Using_systemd_zram-generator to use it.

## Testing done

Built an image (based off current stable Flatcar) that includes `sys-apps/zram-generator`. Running with QEMU created the following config:

```
# /etc/systemd/zram-generator.conf

[zram0]
zram-size = min(ram, 8192)
```

Followed by:

```
$ sudo systemctl daemon-reload
$ sudo systemctl start dev-zram0.swap
```

And verified the swap was created:

```
$ zramctl
NAME       ALGORITHM DISKSIZE DATA COMPR TOTAL STREAMS MOUNTPOINT
/dev/zram0 lzo-rle       962M   4K   80B   12K       6 [SWAP]
```

Big thanks to @jepio with a lot of hand holding that was needed.

- [ ] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)
- [ ] Inspected CI output for image differences: `/boot` and `/usr` size, packages, list files for any missing binaries, kernel modules, config files, kernel modules, etc.

### CODE REVIEW

**Comment by jepio** [comment]
We need to add a keyword for arm64 to the profile, i can provide you a path tomorrow. Also a changelog entry would be nice.

**Review by jepio — APPROVED** [review]

**Comment by jepio** [comment]
I pushed the extra profile keywords. There's a followup required because gentoo rust ebuilds don't handle cross-compilation, the followup is in https://github.com/flatcar/scripts/pull/1780.

Will merge this if the tests run as expected (there is an expected failure from the tang/tpm tests).

**Comment by ader1990** [comment]
Hello,

a mantle CI test might be useful to make sure this new feature has proper regression validation?
Also, I see an addition of static built binaries in the /boot partition, maybe a check on the size increase of the initrd/boot partition would be nice.

Thanks.

**Comment by jepio** [comment]
> Hello,
> 
> a mantle CI test might be useful to make sure this new feature has proper regression validation? Also, I see an addition of static built binaries in the /boot partition, maybe a check on the size increase of the initrd/boot partition would be nice.
> 
> Thanks.

Are you sure? I only see this:
```
diff --git a/tmp/3909.0.0+nightly-20240314-2100-Ztgf0S b/tmp/.-I9wx8L
index 48c3a53899..d89eab3d35 100644
--- a/tmp/3909.0.0+nightly-20240314-2100-Ztgf0S
+++ b/tmp/.-I9wx8L
@@ -20854,6 +20854,7 @@
 /usr/lib/systemd/system-generators/systemd-run-generator
 /usr/lib/systemd/system-generators/systemd-system-update-generator
 /usr/lib/systemd/system-generators/systemd-veritysetup-generator
+/usr/lib/systemd/system-generators/zram-generator
 /usr/lib/systemd/system-preset
 /usr/lib/systemd/system-preset/99-default.preset
 /usr/lib/systemd/system-shutdown
@@ -21350,6 +21351,7 @@
 /usr/lib/systemd/system/systemd-userdbd.socket
 /usr/lib/systemd/system/systemd-vconsole-setup.service
 /usr/lib/systemd/system/systemd-volatile-root.service
+/usr/lib/systemd/system/systemd-zram-setup@.service
 /usr/lib/systemd/system/tcsd.service
 /usr/lib/systemd/system/time-set.target
 /usr/lib/systemd/system/time-sync.target
 ```

Re mantle: lets take care of this separately because it would be a lot of work for a first time contributor and the arm64 package doesn't work yet.

File: sdk_container/src/third_party/portage-stable/changelog/changes/2024-03-21-zram-generator.md [hunk 1]
```diff
@@ -0,0 +1 @@
+- Add zram-generator package to the image ([scripts#1772](https://github.com/flatcar/scripts/pull/1772))
```
[the remark this answers is not in the document]
Comment (george-angel) [reply sdk_container/src/third_party/portage-stable/changelog/changes/2024-03-21-zram-generator.md:1]: Of course, I used this as ref: https://github.com/flatcar/scripts/blob/zram-generator/sdk_container/src/third_party/portage-stable/changelog/changes/2022-10-19-qemu-guest-agent.md#L1 but I did read https://github.com/flatcar/scripts/blob/zram-generator/sdk_container/src/third_party/portage-stable/changelog/README.md#L23-L24 and conveniently forgot. Thanks!

**Comment by ader1990** [comment]
> > Hello,
> > a mantle CI test might be useful to make sure this new feature has proper regression validation? Also, I see an addition of static built binaries in the /boot partition, maybe a check on the size increase of the initrd/boot partition would be nice.
> > Thanks.
> 
> Are you sure? I only see this:
> 
> ```
> diff --git a/tmp/3909.0.0+nightly-20240314-2100-Ztgf0S b/tmp/.-I9wx8L
> index 48c3a53899..d89eab3d35 100644
> --- a/tmp/3909.0.0+nightly-20240314-2100-Ztgf0S
> +++ b/tmp/.-I9wx8L
> @@ -20854,6 +20854,7 @@
>  /usr/lib/systemd/system-generators/systemd-run-generator
>  /usr/lib/systemd/system-generators/systemd-system-update-generator
>  /usr/lib/systemd/system-generators/systemd-veritysetup-generator
> +/usr/lib/systemd/system-generators/zram-generator
>  /usr/lib/systemd/system-preset
>  /usr/lib/systemd/system-preset/99-default.preset
>  /usr/lib/systemd/system-shutdown
> @@ -21350,6 +21351,7 @@
>  /usr/lib/systemd/system/systemd-userdbd.socket
>  /usr/lib/systemd/system/systemd-vconsole-setup.service
>  /usr/lib/systemd/system/systemd-volatile-root.service
> +/usr/lib/systemd/system/systemd-zram-setup@.service
>  /usr/lib/systemd/system/tcsd.service
>  /usr/lib/systemd/system/time-set.target
>  /usr/lib/systemd/system/time-sync.target
> ```
> 
> Re mantle: lets take care of this separately because it would be a lot of work for a first time contributor and the arm64 package doesn't work yet.

looks good to me:

```
File    Size  Used Avail Use% Type
/boot   127M   62M   65M  49% vfat
/usr   1016M  463M  443M  52% btrfs
/       2.0G   36K  1.8G   1% ext4
SUM     3.1G  524M  2.3G  19% -
```

I will add the Mantle test to check if any PR goes beyond the 50% usage in /boot in the following days, so that we won't need to verify manually each time.

**Review by jepio — APPROVED** [review]

**Comment by jepio** [comment]
Thanks for the contribution @george-angel

**Pull request merged**